app/workers/generic_worker.rb

```diff
@@ -0,0 +1,18 @@
+class GenericWorker
+
+  include Sidekiq::Worker
+
+  def perform(klass_name, klass_method, *method_args)
+    klass = klass_name.constantize
+    options = method_args.extract_options!.with_indifferent_access
+    if options
```

Comment:
@vasconsaurus the tests you added in your last commit are still not covering line 15 - I found out that the problem is here... `options` is always a hash, so `if options` is always `true`... you should replace this conditional by something like `unless options.blank?`.

Reply:
@caiosba I updated this, but have a question regarding tests:
The two tests I added were passing before, so they are not really working as expected. How could I fix it?

Also, is there any extra logging or error handling that I should add?

Reply:
Thanks, Manu. No, I don't think any additional logging or error handling is needed here. But for the test, I suggest you actually assess that the method is executed. For now your test for the no-arguments case just verifies that no exception is raised.

Reply:
true 🤦‍♀️ still, do you think I should verify, assert anything else?

Reply:
I updated to use `assert_difference` both here and  for the test in line 28. 
But one last comment: even when the test wasn't hitting the `else`, because I was using `if options`, the test still succeeded. So blank was still created.

So I guess what I mean is, this test makes sure `Blank` is created, but it doesn't make sure of which branch we are hitting.

Reply:
Correct, I think that the other block was still able to handle it just fine. That said, maybe the if/else is not even needed, but it's more readable/clear to have them separate.

Reply:
Yup, so I was wondering about that. I think it might be nice to have the option with just the one straightforward step, but it isn't really needed. Should we keep it or remove it (the conditional)?

Reply:
Yeah, that's what I said, I think it's more clear to have the conditional. Ruby Meta programming can be harder to read, otherwise :)